### DISCUSSION TITLE
brews node and its npm outdated command  suggest a downgrade

### DISCUSSION THREAD

**Discussion by ar4hc** [Homebrew/discussions#258] [Everyday usage]
i installed node (15) using brew and got a npm with it.
The npm was version 7.0.15,
then strange things happened and i got a npm version 6.14.9.

Did a reinstall of node and had npm 7 again.

checked the installed npms and found this:

```bash
# npm -g outdated
Package  Current  Wanted  Latest  Location          Depended by
npm       7.0.15  6.14.9  6.14.9  node_modules/npm  global
```
npm thinks 6 is after 7...

So something "upgraded" my global npm install and got a downgrade instead. Not sure what...
Intellij or some script like depcheck npm-upgrade etc. i think.

the ruby formula did some stuff with the package.json
```ruby
    cp bootstrap/"package.json", libexec/"lib/node_modules/npm"
```

and i think this could be the problem here.

How can i avoid downgrading my npm here?

**Comment by chrmoritz** [comment]
You have to use `npm install -g npm@7` to update your npm, since `npm@6` is still tagged as latest, but node 15 ships with `npm@7`: https://blog.npmjs.org/post/631877012766785536/release-v700

**Comment by klleroy** [comment]
Try running `brew doctor`. That will tell if you there are competing versions of node or something similar. Will also tell you if there are any other issues that need to be addressed. Also, install `nvm` to toggle between node versions as needed.
